### PR TITLE
sopwith: init at 2.3.0

### PR DESCRIPTION
**Pull request by evilbulgarian**
## Description of changes

Adding my info to maintainers and fixing formatting issues thanks to language server.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/sopwith/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, fetchurl
+, glib
+, SDL2
+, libGL
+, pkg-config
+, autoconf
+, automake
```
Comment (a-n-n-a-l-e-e) [inline pkgs/games/sopwith/default.nix:9]:
```suggestion
, autoreconfHook
```

File: pkgs/games/sopwith/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/fragglet/sdl-sopwith/archive/refs/tags/sdl-sopwith-${version}.tar.gz";
+    hash = "sha256-3jIlB9AyWbevtYGEAIlNH6GIUKutEMRtRzay39ByMdg=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    autoconf
+    automake
```
Comment (a-n-n-a-l-e-e) [inline pkgs/games/sopwith/default.nix:24]:
```suggestion
    autoreconfHook
    pkg-config
```

File: pkgs/games/sopwith/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sopwith";
+  version = "2.3.0";
+
+  src = fetchurl {
+    url = "https://github.com/fragglet/sdl-sopwith/archive/refs/tags/sdl-sopwith-${version}.tar.gz";
+    hash = "sha256-3jIlB9AyWbevtYGEAIlNH6GIUKutEMRtRzay39ByMdg=";
+  };
```
Comment (a-n-n-a-l-e-e) [inline pkgs/games/sopwith/default.nix:19]:
```suggestion
  src = fetchFromGitHub {
    owner = "fragglet";
    repo = "sdl-sopwith";
    rev = "refs/tags/sdl-sopwith-${version}";
    hash = "sha256-C0OpFA3q3K5lgoVe0F03/zXlNba/zW7YEIH+2BV/nCI=";
  };
```
fetchurl is not stable fetching this url from github as the hash can change. use fetchFromGitHub instead.

File: pkgs/games/sopwith/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, fetchurl
```
Comment (a-n-n-a-l-e-e) [inline pkgs/games/sopwith/default.nix:3]:
```suggestion
, fetchFromGitHub
```

**Review by ghost — COMMENTED** [review]
Thanks for the PR. there are some things required before it can be ready to merge.

1. remove all of the white space changes made to maintainers.nix
2.  new packages need to be added to [by-name](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#name-based-package-directories) hierarchy. this means move pkgs/games/sopwith/default.nix -> pkgs/by-name/so/sopwith/package.nix
3. fix the review comments
4. verify build `nix-build -A sopwith` from nixpkgs root dir
5. fix commit messages to use the  [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)
briefly, you'll need two commits contained in this PR. one which adds you to the maintainers, and another which adds the new package. it will look like:
    - `maintainers: add evilbulgarian`
    - `sopwith: init at 2.3.0`

File: pkgs/games/sopwith/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    SDL2
+    libGL
+  ];
+
+  preConfigure = "NOCONFIGURE=1 ./autogen.sh";
+
+  meta = with lib; {
+    homepage = "https://github.com/fragglet/sdl-sopwith";
+    description = "Classic biplane shoot ‘em-up game.";
+    license = licenses.gpl2;
```
Comment (a-n-n-a-l-e-e) [inline pkgs/games/sopwith/default.nix:38]:
```suggestion
    license = licenses.gpl2Plus;
```

https://github.com/fragglet/sdl-sopwith/blob/trunk/COPYING.md and the header in various source files states gpl2 or later.

**Comment by a-n-n-a-l-e-e** [comment]
thanks for fixing the commit to `maintainers.nix` but it seems like the commit to add the sopwith package is now missing. it is fine for a PR to contain multiple commits. in this case there should be two:

    maintainers: add evilbulgarian
    sopwith: init at 2.3.0

**Comment by kirillrdy** [comment]
@evilbulgarian @a-n-n-a-l-e-e thank you for your work, I just wanted to adjust PR title to `sopwith: init at 2.3.0` so that its easier to find

**Pull request merged**